BusTracker/Networking/APIResponceParser.swift

```
@@ -9,53 +9,108 @@
 import Foundation
 import Moya
 
+enum APIResponceParserError : Error {
+    case noAPIKey
+    case wrongMethod
+    case noResourceID
+    case unknownError
+    case vehicleParseError(parameter: String)
+}
 
 class APIResponceParser {
     
     func parcePositions(vehicleType : VehicleType,APIResponce : Response) -> [TransportPosition] {
         
         var parcedVehiclePositions: [TransportPosition] = []
         
-        let dateFormatter = DateFormatter()
-        dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
-        
         if let responce = try! JSONSerialization.jsonObject(with: APIResponce.data, options: []) as? [String : Any] {
             
             // Interpret error message from API
             if let message = responce["result"] as? String {
-                switch message{
-                case "false":
-                    print(responce["error"] ?? "Smth wrong with passing an API key")
-                case "Bledna metoda":
-                    print("Wrong type requested. Must be in (1, 2)")
-                case "Błędna metoda lub parametry wywołania":
-                    print("No parameter: resource_id")
-                default:
-                    print("What's going on???")
-                }
+                
+                defineAPIError(result: message, error: responce["error"] as? String)
+                
             } else {
                 
                 let vehiclePositions = responce["result"] as? [Any] ?? []
```

`?? []` is hiding the error about invalid json reply, it should be handled properly. To do that you may want to move this line to `else` statement above. Also you can avoid saving results to the local variable by using early return.

It can look somehow like this:
```swift
if let message = responce["result"] as? String {
    // throws api error
}
else if let vehiclePositions = responce["result"] as? [Any] {
    return positionDataToArray(...)
}
else {
    // throws "wrong json" error
}
```